operator/internal/controller/redpanda/redpanda_controller.go

```diff
@@ -870,6 +874,40 @@ func (r *RedpandaReconciler) reconcileDelete(ctx context.Context, rp *redpandav1
 	return nil
 }
 
+func (r *RedpandaReconciler) deleteHelmChart(ctx context.Context, rp *redpandav1alpha2.Redpanda) error {
+	var hr helmv2beta2.HelmRelease
+	hrName := rp.GetHelmReleaseName()
+
+	if err := r.Client.Get(ctx, client.ObjectKey{Namespace: rp.Namespace, Name: hrName}, &hr); err != nil {
+		// Ignore any error related to retrieving HelmRelease
+		return nil
+	}
+
+	// Try to delete HelmChart custom resource only when HelmRelease is suspended and HelmChart from status is not empty
```

Comment:
This is a re-statement of the `if` clause below, explain why this check is in place.

Kinda a strange edge case in the helm controllers :/

Reply:
Ok adjusted